contrib/utilities/relaunch_simulations.sh

```diff
@@ -0,0 +1,312 @@
+             -st|--set_restart_true: Flag that indicates to the restart argument of FAILED simulations to \"true\" \n
+                                     (deactivated by default, meaning, restart is set to \"false\" in the parameter files.\n
+             "
+
+# Parse named arguments
+while [[ "$#" -gt 0 ]]; do
+  case "$1" in
+    -sf|--search_file)
+      search_file="$2"
+      shift 2
+      ;;
+    -j|--job_file)
+      job_file="$2"
+      shift 2
+      ;;
+    -cp|--copy_search_file)
+      copy_search_file="true"
+      shift
+      ;;
+    -lr|--launch_restart)
+      launch_restart="true"
+      shift
+      ;;
+    -rf|--rerun_failed)
+      rerun_failed="true"
+      shift
+      ;;
+    -st|--set_restart_true)
+      set_restart_true="true"
+      shift
+      ;;
+    -h|--help)
+      echo -e "$USAGE"
+      exit 0
+      ;;
+    *)
+      echo "Unknown parameter: $1"
+      echo -e "$USAGE"
+      exit 1
+      ;;
+  esac
+done
+
+# Assign default values if not set
+search_file="${search_file:-./simulations_list.txt}"
+job_file="${job_file:-launch.sh}"
+copy_search_file="${copy_search_file:-"false"}"
+launch_restart="${launch_restart:-"false"}"
+rerun_failed="${rerun_failed:-"false"}"
+set_restart_true="${set_restart_true:-"false"}"
+
+# Check if the search file exists
+if [ ! -f "$search_file" ]; then
+  echo "File not found: $search_file"
+  exit 1
+fi
+
+# Print values to user
+echo "************************************************************"
+echo -e " Running $0 with parameters: \n"
+echo " Search file:                           $search_file"
+echo " Slurm job file:                        $job_file"
+echo " Copy search file:                      $copy_search_file"
+echo " Launch timeout simulations:            $launch_restart"
+echo " Rerun failed simulations:              $rerun_failed"
+echo " Set restart of failed simulations to:  $set_restart_true"
+echo "************************************************************"
+echo "************************************************************"
+
+# Copy search file if requested otherwise, keep it as a temporary copy.
+# The temporary copy is needed since "relaunched_simulations.txt" is overwritten.
+search_file_prefix=$(basename "$search_file" .txt)
+search_file_copy=$(echo "${search_file_prefix}_old.txt")
+cp "$search_file" "$search_file_copy"
+if [ "$copy_search_file" == "true" ]
+then
+  echo " Search file copied as $search_file_copy"
+  echo "************************************************************"
+fi
+
+################################################################################
+# Function that prints the message when no restart subsection is found
+print_no_restart_message() {
+  local prm_file="$1"
+  echo "No \"subsection restart\" was found in $prm_file"
+  echo "The simulation cannot be restarted."
+  echo "Please add the subsection and relaunch the simulation with checkpointing enabled."
+}
```

Comment:
I am never sure what's the etiquette in bash for where the functions should be defined. Here I think is fine, but it makes me wonder.